x/ibc/03-connection/client/cli/tx.go

```diff
@@ -0,0 +1,507 @@
+			if err != nil {
+				return err
+			}
+
+			var counterpartyPrefix ics23.Prefix
+			if err := cdc.UnmarshalJSON(prefixBz, &counterpartyPrefix); err != nil {
+				return err
+			}
+
+			// TODO: parse strings?
+			counterpartyVersions := args[5]
+
+			proofBz, err := ioutil.ReadFile(args[6])
+			if err != nil {
+				return err
+			}
+
+			var proofInit ics23.Proof
+			if err := cdc.UnmarshalJSON(proofBz, &proofInit); err != nil {
+				return err
+			}
+
+			proofHeight := uint64(cliCtx.Height)
+			consensusHeight, err := lastHeight(cliCtx)
+			if err != nil {
+				return err
+			}
+
+			msg := types.NewMsgConnectionOpenTry(
+				connectionID, clientID, counterpartyConnectionID, counterpartyClientID,
+				counterpartyPrefix, []string{counterpartyVersions}, proofInit, proofHeight,
+				consensusHeight, cliCtx.GetFromAddress(),
+			)
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
+
+// GetCmdConnectionOpenAck defines the command to relay the acceptance of a
+// connection open attempt from chain B to chain A
+func GetCmdConnectionOpenAck(storeKey string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "open-ack [connection-id] [path/to/proof_try.json] [version]",
+		Short: "relay the acceptance of a connection open attempt from chain B to chain A",
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`relay the acceptance of a connection open attempt from chain B to chain A:
+
+Example:
+$ %s tx ibc connection open-ack [connection-id] [path/to/proof_try.json] [version]
+		`, version.ClientName),
+		),
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().WithCodec(cdc)
+			// cliCtx2 := context.NewCLIContextWithFrom(viper.GetString(FlagFrom2)).
+			// 	WithCodec(cdc).
+			// 	WithNodeURI(viper.GetString(FlagNode2)).
+			// 	WithBroadcastMode(flags.BroadcastBlock)
+
+			connectionID := args[0]
+			proofBz, err := ioutil.ReadFile(args[1])
+			if err != nil {
+				return err
+			}
+
+			var proofTry ics23.Proof
+			if err := cdc.UnmarshalJSON(proofBz, &proofTry); err != nil {
+				return err
+			}
+
+			proofHeight := uint64(cliCtx.Height)
+			consensusHeight, err := lastHeight(cliCtx)
+			if err != nil {
+				return err
+			}
+
+			version := args[4]
+
+			msg := types.NewMsgConnectionOpenAck(
+				connectionID, proofTry, proofHeight,
+				consensusHeight, version, cliCtx.GetFromAddress(),
+			)
+
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
+
+// GetCmdConnectionOpenConfirm defines the command to initialize a connection on
+// chain A with a given counterparty chain B
+func GetCmdConnectionOpenConfirm(storeKey string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "open-confirm [connection-id] [path/to/proof_ack.json]",
```

Comment:
How to query for `path/to/proof_ack.json`?

Reply:
when you query the connection, the response should include the latest `ics23.Proof` as well. Or alternative a list of all the previous proofs from each of the prev steps done (i.e init, opentry, etc).